src/main/java/seedu/address/logic/parser/EditCommandParser.java

```diff
@@ -39,7 +39,11 @@ public EditCommand parse(String args) throws ParseException {
         Index index;
 
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            if (argMultimap.getPreamble().isEmpty()) {
+                index = null;
+            } else {
+                index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            }
```

Comment:
this looks a little nested but there's no easy way to fix it. a way i can think of is to put this in another class so that guard clauses can be used

```java
Index parse(ArgumentMultimap argMultimap) {
    if (argMultimap.getPreamble().isEmpty()) {
        return null;
    }
    return ParserUtil.parseIndex(argMultimap.getPreamble());
}

...
try {
    Index index = parse(argMultimap);
} catch (ParseException pe) {
    throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE), pe);
}
...

```

Reply:
I agree with this, though parse will have to throw the `ParseException`, thanks for your suggestion.